### PR TITLE
swap risk assessment page

### PR DESCRIPTION
**Pull request by StupidCodeFactory**
## Description
surgically introduce new risk assessment form into the trading standard. 
I've _tried_ to follow how things works with *other_types_information* but separate the `:risk_assessments` step in own switches and `(before|after)_action` `store_risk_assessment && set_risk_assessment`. 
State management is done in the session and file blobs id are stored alongside its respective risk_assessment  attributes in the session.

## Checklist:
- [X] Have you documented your changes in the pull request description?
- [ ] Has acceptance criteria been tested by a peer?
- [ ] Has the CHANGELOG been updated? (If change is worth telling users about.)

### CODE REVIEW

**Comment by StupidCodeFactory** [comment]
[(5) Swap risk assessment pages in trading standards add flow](https://trello.com/c/IJ2BmtMF/606-5-swap-risk-assessment-pages-in-trading-standards-add-flow)

File: config/redis_session.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 default: &default
-  url: <%= ENV.fetch('REDIS_URL', 'redis://redis:6379') %>
+  url: <%= ENV.fetch('REDIS_URL', 'redis://localhost:6379') %>
```
Comment (StupidCodeFactory) [inline config/redis_session.yml:2]: I could remove this. somehow my .env file is not picked up anymore and as we do not use docker-compose I've just defaulted its value to localhost.

**Review by nasirkhanbeis — APPROVED** [review]
looks good as expected!
Just wondering if we want to include further step to update case level risk if user chosen to add risk assessment in case creation flow.

File: app/controllers/investigations/ts_investigations_controller.rb [hunk 2]
```diff
@@ -163,6 +168,11 @@ def set_repeat_step(model = :investigation)
                    end
   end
 
+  def set_risk_assessment_form
+    @risk_assessment_form = TradingStandardRiskAssessmentForm.new(current_user: current_user, investigation: @investigation, businesses: businesses_from_session, product: @product)
+    set_repeat_step(:trading_standard_risk_assessment_form)
+  end
```
Comment (slorek) [inline app/controllers/investigations/ts_investigations_controller.rb:174]: Trading Standards is always plural:
```suggestion
  def set_risk_assessment_form
    @risk_assessment_form = TradingStandardsRiskAssessmentForm.new(current_user: current_user, investigation: @investigation, businesses: businesses_from_session, product: @product)
    set_repeat_step(:trading_standards_risk_assessment_form)
  end
```

File: app/controllers/investigations/ts_investigations_controller.rb [hunk 3]
```diff
@@ -494,6 +523,34 @@ def records_valid?
...
+      if (file = trading_standard_risk_assessment_form_params[:risk_assessment_file])
+        @file_blob = ActiveStorage::Blob.create_after_upload!(
+          io: file,
+          filename: file.original_filename,
+          content_type: file.content_type,
+          metadata: {
+            created_by: current_user.id
+          }
+        )
+      end
```
Comment (slorek) [inline app/controllers/investigations/ts_investigations_controller.rb:548]:
```suggestion
    when :risk_assessments
      @risk_assessment_form = TradingStandardsRiskAssessmentForm.new(
        trading_standards_risk_assessment_form_params
          .merge(
            current_user: current_user,
            investigation: @investigation,
            businesses: businesses_from_session,
            product: @product
          )
      )

      @investigation = @investigation.decorate

      if (file = trading_standards_risk_assessment_form_params[:risk_assessment_file])
        @file_blob = ActiveStorage::Blob.create_after_upload!(
          io: file,
          filename: file.original_filename,
          content_type: file.content_type,
          metadata: {
            created_by: current_user.id
          }
        )
      end
```

File: app/controllers/investigations/ts_investigations_controller.rb [hunk 4]
```diff
@@ -510,6 +567,14 @@ def records_valid?
     @investigation.errors.empty? && @product.errors.empty?
   end
 
+  def trading_standard_risk_assessment_form_params
+    params.require(:trading_standard_risk_assessment_form).permit(:further_risk_assessments, :details, :risk_level, :risk_assessment_file, :assessed_by, :assessed_by_team_id, :assessed_by_business_id, :assessed_by_other, :custom_risk_level, assessed_on: %i[day month year], product_ids: [])
+  end
```
Comment (slorek) [inline app/controllers/investigations/ts_investigations_controller.rb:572]:
```suggestion
  def trading_standards_risk_assessment_form_params
    params.require(:trading_standards_risk_assessment_form).permit(:further_risk_assessments, :details, :risk_level, :risk_assessment_file, :assessed_by, :assessed_by_team_id, :assessed_by_business_id, :assessed_by_other, :custom_risk_level, assessed_on: %i[day month year], product_ids: [])
  end
```

File: app/forms/trading_standard_risk_assessment_form.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+class TradingStandardRiskAssessmentForm < RiskAssessmentForm
```
Comment (slorek) [inline app/forms/trading_standard_risk_assessment_form.rb:1]:
```suggestion
class TradingStandardsRiskAssessmentForm < RiskAssessmentForm
```

**Review by slorek — APPROVED** [review]
Looks good!

**Pull request merged**